core/src/main/java/org/apache/carbondata/core/datamap/TableDataMap.java

```diff
@@ -167,8 +167,9 @@ public CarbonTable getTable() {
         }
       } else {
         for (DataMap dataMap : dataMaps.get(segment)) {
+          // TODO: just for test will be reverted back after test
           pruneBlocklets.addAll(
-              dataMap.prune(filter.getExpression(), segmentProperties, partitions, table));
+              dataMap.prune(filter.getNewCopyOfExpression(), segmentProperties, partitions, table));
```

Comment:
please revert to old code

Reply:
done. It was added to run CI. removed now